[ci-visibility] Add known limitation around jest's `forceExit` option (#21798)
DataDog/documentation: content/en/tests/setup/javascript.md

### Jest's `test.concurrent`
Jest's [test.concurrent][17] is not supported.

### Jest's `--forceExit`

> Jest's [--forceExit][21] option might cause data to be lost. We try to send data as soon as possible after your tests finish, but shutting down the process abruptly can cause some requests to fail. Use `--forceExit` with caution.

Reviewer: ```suggestion
Jest's [--forceExit][21] option may cause data loss. Datadog tries to send data immediately after your tests finish, but shutting down the process abruptly can cause some requests to fail. Use `--forceExit` with caution.
```
